metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -754,15 +750,21 @@ def generate_container_for_workspace(
 
     def generate_container_for_dataset(
         self, dataset: powerbi_data_classes.PowerBIDataset
-    ) -> Iterable[MetadataWorkUnit]:
+    ) -> Iterable[MetadataChangeProposalWrapper]:
         dataset_key = dataset.get_dataset_key(self.__config.platform_name)
         container_work_units = gen_containers(
             container_key=dataset_key,
             name=dataset.name if dataset.name else dataset.id,
             parent_container_key=self.workspace_key,
             sub_types=[BIContainerSubTypes.POWERBI_DATASET],
         )
-        return container_work_units
+
+        # The if statement here is just to satisfy mypy
+        return [
+            wu.metadata
+            for wu in container_work_units
+            if isinstance(wu.metadata, MetadataChangeProposalWrapper)
+        ]
```

Comment:
This is kind of ugly, but I couldn't find any other implementation for `gen_containers` and decided that for this implementation it's easier to unwrap the metadata from the work unit rather than create another implementation for container creation.

Reply:
this is fine - we should probably make gen_containers return MCPs instead of workunits